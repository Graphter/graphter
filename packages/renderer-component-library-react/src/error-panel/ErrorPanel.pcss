.error {
  background-color: #ff00490d;
  border: 3px dotted #ff0049;
  padding: 30px;
  margin: 30px 0;
}