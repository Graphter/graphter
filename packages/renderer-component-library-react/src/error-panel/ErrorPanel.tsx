import React from 'react';
import s from './ErrorPanel.pcss';

interface ErrorPanelProps {
  err: Error | string
}

export default function ErrorPanel({ err }: ErrorPanelProps){
  let message: string = 'An error occurred';
  if(typeof err === 'string') message = err;
  else if(err && 'message' in err && err.message) message = err.message;
  return (
    <div className={s.error} data-testid='error-panel'>{message}</div>
  )
}