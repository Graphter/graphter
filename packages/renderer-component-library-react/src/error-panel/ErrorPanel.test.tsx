import { act, render, within, fireEvent } from '@testing-library/react';
import { screen } from '@testing-library/dom'
import React from "react";
import ErrorPanel from "./ErrorPanel";

describe(`<ErrorPanel />`, () => {
  it(`should display an Error error message`, () => {
    render(<ErrorPanel err={new Error('some Error error message' )} />);
    const errorPanel = screen.getByTestId('error-panel');
    expect(errorPanel.textContent).toMatchSnapshot();
  });
  it(`should display a string error message`, () => {
    render(<ErrorPanel err={'some string error message'} />);
    const errorPanel = screen.getByTestId('error-panel');
    expect(errorPanel.textContent).toMatchSnapshot();
  });
  it(`should display a default message if no error is provided`, () => {
    // @ts-ignore
    render(<ErrorPanel err={null} />);
    const errorPanel = screen.getByTestId('error-panel');
    expect(errorPanel.textContent).toMatchSnapshot();
  });
  it(`should display a default message if no error message is provided`, () => {
    render(<ErrorPanel err={new Error()} />);
    const errorPanel = screen.getByTestId('error-panel');
    expect(errorPanel.textContent).toMatchSnapshot();
  });
});

export {};