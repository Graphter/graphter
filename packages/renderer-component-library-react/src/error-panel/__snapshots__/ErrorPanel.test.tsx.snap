// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<ErrorPanel /> should display a default message if no error is provided 1`] = `"An error occurred"`;

exports[`<ErrorPanel /> should display a default message if no error message is provided 1`] = `"An error occurred"`;

exports[`<ErrorPanel /> should display a string error message 1`] = `"some string error message"`;

exports[`<ErrorPanel /> should display an Error error message 1`] = `"some Error error message"`;

exports[`<ErrorPanel /> should display the error message 1`] = `"some error message"`;
